packages/wdio-spec-reporter/src/index.ts

```diff
@@ -181,15 +181,25 @@ export default class SpecReporter extends WDIOReporter {
         )
 
         if (isSauceJob && config && config.user && config.key && sessionId) {
-            const isUSEast = config.headless || (config.hostname?.includes('us-east-1'))
-            const isEUCentral = ['eu', 'eu-central-1'].includes(config?.region || '') || (config.hostname?.includes('eu-central'))
-            const isAPAC = ['apac', 'apac-southeast-1'].includes(config?.region || '') || (config.hostname?.includes('apac'))
-            const dc = isUSEast ? '.us-east-1' : isEUCentral ? '.eu-central-1' : isAPAC ? '.apac-southeast-1' : ''
+            let sauceUrl
             const multiremoteNote = isMultiremote ? ` ${instanceName}` : ''
-            const sauceLabsSharableLinks = this._sauceLabsSharableLinks
-                ? sauceAuthenticationToken( config.user, config.key, sessionId )
-                : ''
-            return [`Check out${multiremoteNote} job at https://app${dc}.saucelabs.com/tests/${sessionId}${sauceLabsSharableLinks}`]
+
+            // The report url of RDC is in the caps that are returned
+            if ('testobject_test_report_url' in capabilities){
+                sauceUrl = capabilities.testobject_test_report_url
+            } else {
+                // VDC urls can be constructed / be made shared
+                const isUSEast = config.headless || (config.hostname?.includes('us-east-1'))
+                const isEUCentral = ['eu', 'eu-central-1'].includes(config?.region || '') || (config.hostname?.includes('eu-central'))
+                const isAPAC = ['apac', 'apac-southeast-1'].includes(config?.region || '') || (config.hostname?.includes('apac'))
+                const dc = isUSEast ? '.us-east-1' : isEUCentral ? '.eu-central-1' : isAPAC ? '.apac-southeast-1' : ''
+                const sauceLabsSharableLinks = this._sauceLabsSharableLinks
+                    ? sauceAuthenticationToken( config.user, config.key, sessionId )
+                    : ''
+                sauceUrl = `https://app${dc}.saucelabs.com/tests/${sessionId}${sauceLabsSharableLinks}`
+            }
+
+            return [`Check out${multiremoteNote} job at ${sauceUrl}`]
```

Comment:
Let's return early rather than having an if/else
```suggestion
            const note = 'Check out %s job at %s'

            // The report url of RDC is in the caps that are returned
            if ('testobject_test_report_url' in capabilities){
                return util.format(note, multiremoteNote, capabilities.testobject_test_report_url)
            }
    
            // VDC urls can be constructed / be made shared
            const isUSEast = config.headless || (config.hostname?.includes('us-east-1'))
            const isEUCentral = ['eu', 'eu-central-1'].includes(config?.region || '') || (config.hostname?.includes('eu-central'))
            const isAPAC = ['apac', 'apac-southeast-1'].includes(config?.region || '') || (config.hostname?.includes('apac'))
            const dc = isUSEast ? '.us-east-1' : isEUCentral ? '.eu-central-1' : isAPAC ? '.apac-southeast-1' : ''
            const sauceLabsSharableLinks = this._sauceLabsSharableLinks
                ? sauceAuthenticationToken( config.user, config.key, sessionId )
                : ''
            sauceUrl = `https://app${dc}.saucelabs.com/tests/${sessionId}${sauceLabsSharableLinks}`
            return [`Check out ${multiremoteNote} job at ${sauceUrl}`]
```

Reply:
Thanks @christian-bromann 

Quick question. where is `util` coming from?

Reply:
```ts
import util from 'util'`
```
or
```
import { format } from 'util'
```

Reply:
NVM, it's the util native lib

Reply:
Done, I've pushed the new code, thanks for the feedback